### PR TITLE
WEBAPP-568: Show error for disabled news

### PR DESCRIPTION
**Pull request by NKlug**
We now show a generic error if push-news/tu-news are disabled. I also adjusted the page title and fixed a small bug with local-news paths in the LocalNewsList.

This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/modules/app/route-configs/TunewsRouteConfig.js [hunk 1]
```diff
@@ -29,7 +32,11 @@ const tunewsRoute: Route = {
       fetchData(createCitiesEndpoint(cmsApiBaseUrl), dispatch, state.cities),
       fetchData(createEventsEndpoint(cmsApiBaseUrl), dispatch, state.events, { city, language }),
       fetchData(createTunewsLanguagesEndpoint(tunewsApiBaseUrl, true), dispatch, state.languages, { city, language }),
-      fetchData(createTunewsEndpoint(tunewsApiBaseUrl), dispatch, state.tunews, { page: 1, language: language, count: 20 })
+      fetchData(createTunewsEndpoint(tunewsApiBaseUrl), dispatch, state.tunews, {
```
Comment (maxammann) [inline src/modules/app/route-configs/TunewsRouteConfig.js:35]: Try to avoid these changes as no linting rule was violated. If the line was too long for the linter then this is fine though :+1:

File: src/routes/news/containers/LocalNewsPage.js [hunk 2]
```diff
@@ -49,12 +49,11 @@ export class LocalNewsPage extends React.Component<PropsType> {
     }
 
     const currentCity: CityModel = cities && cities.find(cityElement => cityElement.code === city)
-
     if (!currentCity.pushNotificationsEnabled) {
-      const type = currentCity.tunewsEnabled ? 'tunewsItem' : 'category'
-      const error = new ContentNotFoundError({ type, id: path, city: city, language })
+      const error = new ContentNotFoundError({ type: 'category', id: path, city: city, language })
```
Comment (maxammann) [inline src/routes/news/containers/LocalNewsPage.js:53]: Why are we using here 'category' instead of 'tunews'?

File: src/routes/news/containers/TunewsDetailsPage.js [hunk 3]
```diff
@@ -55,15 +55,18 @@ type PropsType = {|
...
 export class TunewsDetailsPage extends React.PureComponent<PropsType> {
   render () {
-    const { tunewsElement, language, path, city } = this.props
+    const { tunewsElement, language, path, city, cities } = this.props
 
-    if (!tunewsElement) {
-      const error = new ContentNotFoundError({ type: 'tunewsItem', id: path, city, language })
+    const currentCity: CityModel = cities && cities.find(cityElement => cityElement.code === city)
+    if (!tunewsElement || !currentCity.tunewsEnabled) {
+      const type: string = currentCity.tunewsEnabled ? 'tunewsItem' : 'category'
```
Comment (maxammann) [inline src/routes/news/containers/TunewsDetailsPage.js:68]: In the LocalNewsPage.js we always choose 'category'. Which one should we use here?

File: src/routes/news/containers/TunewsPage.js [hunk 4]
```diff
@@ -60,10 +60,8 @@ export class TunewsPage extends React.PureComponent<PropsType> {
     }
 
     const currentCity: CityModel = cities && cities.find(cityElement => cityElement.code === city)
-
     if (!currentCity.tunewsEnabled) {
-      const type = currentCity.pushNotificationsEnabled ? 'localNewsItem' : 'category'
-      const error = new ContentNotFoundError({ type, id: path, city: city, language })
+      const error = new ContentNotFoundError({ type: 'category', id: path, city: city, language })
```
Comment (maxammann) [inline src/routes/news/containers/TunewsPage.js:64]: Same as above, not sure which to choose.

File: src/routes/news/containers/LocalNewsPage.js (hunk 2 above)
Comment (klinzo) [reply src/routes/news/containers/LocalNewsPage.js:53]: Imo it should be `category` all the time. If a feature is disabled, we want to show a generic error (which is category in our case). I think the same applies for the questions below.

File: src/modules/app/route-configs/TunewsRouteConfig.js (hunk 1 above)
Comment (NKlug) [reply src/modules/app/route-configs/TunewsRouteConfig.js:35]: Ok. Just hit autoformat I guess...

File: src/routes/news/containers/LocalNewsPage.js (hunk 2 above)
Comment (NKlug) [reply src/routes/news/containers/LocalNewsPage.js:53]: Yep, @klinzo is right. Changing this was the main goal of this task.

File: src/routes/news/containers/TunewsDetailsPage.js (hunk 3 above)
Comment (NKlug) [reply src/routes/news/containers/TunewsDetailsPage.js:68]: If the tu-news feature is enabled, I'd show an error that the corresponding tu-news-Element does not exist.
If it's disabled, this route is invalid, so imo a generic CategoryNotFoundError should be shown.

Comment (NKlug) [reply src/routes/news/containers/TunewsDetailsPage.js:68]: This will be reworked in WEBAPP-569 anyways. So it's not that important I think.

File: src/modules/app/route-configs/TunewsRouteConfig.js (hunk 1 above)
Comment (klinzo) [reply src/modules/app/route-configs/TunewsRouteConfig.js:35]: But I guess the line was too long, so it should be fine, right?

File: src/routes/news/containers/TunewsDetailsPage.js [hunk 5]
```diff
@@ -55,15 +55,18 @@ type PropsType = {|
...
+      const error = new ContentNotFoundError({ type, id: path, city, language })
       return <FailureSwitcher error={error} />
     }
```
Comment (klinzo) [inline src/routes/news/containers/TunewsDetailsPage.js:71]:
```suggestion
    if (!currentCity.tunewsEnabled) {
      const error = new ContentNotFoundError({ type: 'category', id: path, city, language })
      return <FailureSwitcher error={error} />
    } else if (!tunewsElement) {
    ...
    }
```
Much clearer imo.

File: src/routes/news/components/NewsElement.js [hunk 6]
```diff
@@ -74,7 +74,7 @@ class NewsElement extends React.PureComponent<PropsType> {
 
   render () {
     const { path, id } = this.props
-    const itemPath = `${path}/${id}`
+    const itemPath = `${path}${id}`
```
Comment (klinzo) [inline src/routes/news/components/NewsElement.js:77]: Path already contains a slash or why this change?

File: src/routes/news/containers/TunewsPage.js (hunk 4 above)
Comment (klinzo) [reply src/routes/news/containers/TunewsPage.js:64]: This is right imo.

File: src/routes/news/containers/LocalNewsDetailsPage.js [hunk 7]
```diff
@@ -15,15 +15,18 @@ type PropsType = {|
...
+    const { localNewsElement, language, city, path, cities } = this.props
 
-    if (!localNewsElement) {
-      const error = new ContentNotFoundError({ type: 'localNewsItem', id: path, city, language })
+    const currentCity: CityModel = cities && cities.find(cityElement => cityElement.code === city)
+    if (!localNewsElement || !currentCity.pushNotificationsEnabled) {
+      const type: string = currentCity.pushNotificationsEnabled ? 'localNewsItem' : 'category'
+      const error = new ContentNotFoundError({ type, id: path, city, language })
       return <FailureSwitcher error={error} />
     }
```
Comment (klinzo) [inline src/routes/news/containers/LocalNewsDetailsPage.js:31]: I'd rather split in if else if like in TunewsDetailsPage. But your choice.

File: src/routes/news/containers/TunewsDetailsPage.js (hunk 3 above)
Comment (klinzo) [reply src/routes/news/containers/TunewsDetailsPage.js:68]: @maxammann this is the detail page, so it is correctly done imo and has the same behavior as the `LocalNewsDetailsPage`

Comment (klinzo) [reply src/routes/news/containers/TunewsDetailsPage.js:68]: @NKlug why do you think this will be reworked in `WEBAPP-569`?

File: src/routes/news/components/NewsElement.js (hunk 6 above)
Comment (NKlug) [reply src/routes/news/components/NewsElement.js:77]: Yep. It wasn't working with the slash.

Comment (klinzo) [reply src/routes/news/components/NewsElement.js:77]: What exactly was not working? It's working for me (to click on a news element in the news list).

File: src/routes/news/containers/TunewsDetailsPage.js (hunk 3 above)
Comment (NKlug) [reply src/routes/news/containers/TunewsDetailsPage.js:68]: I guess a didn't read the whole thing properly. I thought it involved redirecting to the TunewsPage if the TunewsElement has not been found/does not exist.

File: src/routes/news/components/NewsElement.js (hunk 6 above)
Comment (NKlug) [reply src/routes/news/components/NewsElement.js:77]: @markl told me he already fixed this somewhere else. I'll remove it.

File: src/routes/news/containers/LocalNewsDetailsPage.js (hunk 7 above)
Comment (NKlug) [reply src/routes/news/containers/LocalNewsDetailsPage.js:31]: Done.

File: src/routes/news/containers/TunewsDetailsPage.js (hunk 5 above)
Comment (NKlug) [reply src/routes/news/containers/TunewsDetailsPage.js:71]: Done.

**Comment by Taggotty** [comment]
Can you check whether this branch resolves [WEBAPP-571](https://issues.integreat-app.de/browse/WEBAPP-571)?

File: src/routes/news/containers/LocalNewsDetailsPage.js [hunk 8]
```diff
@@ -3,26 +3,34 @@
...
 
 export class LocalNewsDetailsPage extends React.PureComponent<PropsType> {
   render () {
-    const { localNewsElement, language, city, id } = this.props
+    const { localNewsElement, language, city, cities, id } = this.props
 
-    if (!localNewsElement) {
+    const currentCity: ?CityModel = cities && cities.find(cityElement => cityElement.code === city)
+    if (!currentCity) {
+      return <FailureSwitcher error={new CityNotFoundError()} />
```
[the remark this answers is not in the document]
Comment (NKlug) [reply src/routes/news/containers/LocalNewsDetailsPage.js:29]: Yep

**Comment by NKlug** [comment]
@Taggotty This doesn't resolve WEBAPP-571, but I came across the same issue. This has to do with the error returned from the Apiclient. I created corresponding tasks to fix this (APICLIENT-37 and WEBAPP-579). Should we close WEBAPP-571?
